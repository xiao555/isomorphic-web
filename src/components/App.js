import React from 'react'
import PropTypes from 'prop-types'

const ContextType = {
  // Enables critical path CSS rendering
  // https://github.com/kriasoft/isomorphic-style-loader
  insertCss: PropTypes.func.isRequired,
  // Universal HTTP client
  fetch: PropTypes.func.isRequired,
  pathname: PropTypes.string.isRequired,
  query: PropTypes.object,
}

class App extends React.PureComponent {
  static propTypes = {
    context: PropTypes.shape(ContextType).isRequired,
    children: PropTypes.element.isRequired,
  }

  static childContextTypes = ContextType

  getChildContext() {
    return this.props.context
  }

  render() {
    return React.Children.only(this.props.children)
  }
}

export default App