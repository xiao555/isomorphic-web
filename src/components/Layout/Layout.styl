@import '../variables.styl'

html
  color #222
  font-weight 100
  font-size 1em // ~16px
  font-family $font-family-base

body
  marign 0

a 
 color #0074c2

