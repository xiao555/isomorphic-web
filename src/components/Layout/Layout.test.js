import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import Layout from './Layout';

describe('Test Layout', () => {
  test('renders children correctly', () => {
    const wrapper = renderer
      .create(
        <App context={{ insertCss: () => {}, fetch: () => {}, pathname: '' }}>
          <Layout>
            <div className="child" />
          </Layout>
        </App>,
      )
      .toJSON();
    
    expect(wrapper).toMatchSnapshot();
  });
});
