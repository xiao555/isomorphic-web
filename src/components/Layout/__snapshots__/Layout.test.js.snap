// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Layout renders children correctly 1`] = `
<div>
  <header>
    <div
      className="container"
    >
      <div
        className="root"
        role="navigation"
      >
        <a
          className="link"
          href="./home"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="link"
          href="./about"
          onClick={[Function]}
        >
          About
        </a>
      </div>
      <a
        className="brand"
        href="/"
        onClick={[Function]}
      >
        <span
          className="brandTxt"
        >
          Logo
        </span>
      </a>
    </div>
    <div
      className="banner"
    >
      <h1
        className="bannerTitle"
      >
        Isomorphic Web App
      </h1>
      <a
        className="bannerLink"
        href="https://github.com/xiao555/isomorphic-web"
        target="_blank"
      >
        Github
      </a>
    </div>
  </header>
  <div
    className="child"
  />
  <footer
    className="root"
  >
    <div
      className="container"
    >
      <span
        className="text"
      >
        © Xiao555
      </span>
      <span
        className="spacer"
      >
        ·
      </span>
      <a
        className="link"
        href="https://github.com/xiao555"
        target="_blank"
      >
        Github
      </a>
    </div>
  </footer>
</div>
`;
